telegram/ext/commandhandler.py

```diff
@@ -110,6 +115,15 @@ def __init__(self,
             self.command = [x.lower() for x in command]
         self.filters = filters
         self.allow_edited = allow_edited
+        if prefix is None or not isinstance(prefix, string_types) or len(prefix) > 1:
+            raise ValueError("The prefix argument to CommandHandler must be a single character or "
+                             "an empty string.")
+        if prefix == ' ':
+            # A leading space character is stripped away in Telegram messages, so we can consider
+            # it as an empty string prefix.
+            self.prefix = ''
+        else:
+            self.prefix = prefix
```

Comment:
* I don't think prefix should be limited to a single char. Same as you accept an empty string you can accept any string longer than 1 char.

* Instead of calculating (over and over) the `empty_leader` during `check_update`, you can calculate `self.leader_len` here (at the c'tor) and simplify everything else later on.

* Calling `len(prefix)` here will also make the check for `None` redundant. It will be the one to raise `TypeError` if necessary.